### PR TITLE
Change coherence workload wrapper for istio

### PR DESCRIPTION
**Pull request by bobdonat**
# Description

The PR creates/updates the Istio destination rule for Coherence workloads when the namespace is istio enabled.
In addition, changes were made to the WebLogic workload to only create the Istio destination rule if it does not exist.

Fixes VZ-2239

# Checklist 

As the author of this PR, I have:

- [x] Checked that I included or updated copyright and license notices in all files that I altered
- [x] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by ddrechse — APPROVED** [review]

File: application-operator/controllers/cohworkload/coherenceworkload_controller.go [hunk 1]
```diff
@@ -382,3 +399,75 @@ func addJvmArgs(coherenceSpec map[string]interface{}) {
...
+	destinationRule.Spec.Host = fmt.Sprintf("*.%s.svc.cluster.local", namespace)
+	destinationRule.Spec.TrafficPolicy = &istionet.TrafficPolicy{
+		Tls: &istionet.ClientTLSSettings{
+			Mode: istionet.ClientTLSSettings_ISTIO_MUTUAL,
+		},
+	}
+	destinationRule.Spec.TrafficPolicy.PortLevelSettings = []*istionet.TrafficPolicy_PortTrafficPolicy{
+		{
+			Port: &istionet.PortSelector{
+				Number: 9000,
```
Comment (kminder) [inline application-operator/controllers/cohworkload/coherenceworkload_controller.go:453]: What is this port for any why is it ok that it is hard coded to 9000.  Seems like a comment explaining that would be a good idea.

File: application-operator/controllers/cohworkload/coherenceworkload_controller_test.go [hunk 2]
```diff
@@ -472,6 +494,136 @@ func TestReconcileFetchWorkloadError(t *testing.T) {
...
+				Kind:       "ApplicationConfiguration",
+			}
+			return nil
+		})
+
+	// Expect a call to create the destinationRule and return success
+	cli.EXPECT().
+		Create(gomock.Any(), gomock.Any()).
+		DoAndReturn(func(ctx context.Context, dr *istioclient.DestinationRule, opts ...client.CreateOption) error {
+			return nil
```
Comment (kminder) [inline application-operator/controllers/cohworkload/coherenceworkload_controller_test.go:527]: Seems like there should be some assertions in here or is it enough to just check that the function was called?

File: application-operator/controllers/cohworkload/coherenceworkload_controller_test.go [hunk 3]
```diff
@@ -472,6 +494,136 @@ func TestReconcileFetchWorkloadError(t *testing.T) {
...
+				Kind:       "ApplicationConfiguration",
+			}
+			return nil
+		})
+
+	// Expect a call to update the destinationRule and return success
+	cli.EXPECT().
+		Update(gomock.Any(), gomock.Any()).
+		DoAndReturn(func(ctx context.Context, dr *istioclient.DestinationRule, opts ...client.CreateOption) error {
+			return nil
```
Comment (kminder) [inline application-operator/controllers/cohworkload/coherenceworkload_controller_test.go:580]: Assertions?

File: application-operator/controllers/cohworkload/coherenceworkload_controller_test.go [hunk 4]
```diff
@@ -472,6 +494,136 @@ func TestReconcileFetchWorkloadError(t *testing.T) {
...
+// WHEN the controller createOrUpdateDestinationRule function is called
+// THEN expect an error to be returned
+func TestCreateUpdateDestinationRuleNoLabel(t *testing.T) {
+	assert := asserts.New(t)
+
+	reconciler := Reconciler{}
+	namespaceLabels := make(map[string]string)
+	workloadLabels := make(map[string]string)
+	err := reconciler.createOrUpdateDestinationRule(context.Background(), ctrl.Log, "test-namespace", namespaceLabels, workloadLabels)
+	assert.Nil(err)
```
Comment (kminder) [inline application-operator/controllers/cohworkload/coherenceworkload_controller_test.go:624]: This doesn't match what the comment says this test done.

File: application-operator/controllers/cohworkload/coherenceworkload_controller.go (hunk 1 above)
Comment (bobdonat) [reply application-operator/controllers/cohworkload/coherenceworkload_controller.go]: added comment

File: application-operator/controllers/cohworkload/coherenceworkload_controller_test.go (hunk 2 above)
Comment (bobdonat) [reply application-operator/controllers/cohworkload/coherenceworkload_controller_test.go:576]: Added assertions.

File: application-operator/controllers/cohworkload/coherenceworkload_controller_test.go (hunk 3 above)
Comment (bobdonat) [reply application-operator/controllers/cohworkload/coherenceworkload_controller_test.go:638]: Fixed

File: application-operator/controllers/cohworkload/coherenceworkload_controller_test.go (hunk 4 above)
Comment (bobdonat) [reply application-operator/controllers/cohworkload/coherenceworkload_controller_test.go]: Fixed.

**Review by kminder — APPROVED** [review]

**Pull request merged**